### PR TITLE
fix(eap) Separate the RPC code from the other snuba code

### PR DESCRIPTION
**Pull request by evanh**
Move the RPC endpoint and code into a separate file. Also create a new threadpool for the RPC
to use. This won't result in more work, queries are sent either to the RPC or the SnQL endpoint
but not both.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Review by volokluev — APPROVED** [review]

**Review by shellmayr — APPROVED** [review]

**Review by colin-sentry — CHANGES_REQUESTED** [review]

File: src/sentry/utils/snuba_rpc.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        attribute_key_transform_context=AttributeKeyTransformContext(),
+    )
+    aggregate_resp = snuba.rpc(aggregate_req, AggregateBucketResponse)
+    """
+    referrer = req.meta.referrer
+    with sentry_sdk.start_span(op="snuba_rpc.run", description=req.__class__.__name__) as span:
+        span.set_tag("snuba.referrer", referrer)
+        http_resp = _snuba_pool.urlopen(
+            "POST",
+            f"/rpc/{req.__class__.__name__}",
```
[the remark this answers is not in the document]
Comment (evanh) [reply src/sentry/utils/snuba_rpc.py:78]: I copied this straight from utils/snuba.py, and I don't see the new URL anywhere. What is this supposed to be?

[the remark this answers is not in the document]
Comment (colin-sentry) [reply src/sentry/utils/snuba_rpc.py:78]: /rpc/AggregateBucketRequest/v1alpha for example

`api_version = req.__class__.__module__.split('.')[-1]` I think

**Review by colin-sentry — APPROVED** [review]

File: src/sentry/utils/snuba_rpc.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zylphrex) [reply src/sentry/utils/snuba_rpc.py]: feel free to leave alone for now as i have the fix in https://github.com/getsentry/sentry/pull/78469

[the remark this answers is not in the document]
Comment (volokluev) [reply src/sentry/utils/snuba_rpc.py]: good catch colin, yes there is a version now as well

**Pull request merged**